### PR TITLE
Objdetect: QRCode detect + decode

### PR DESCRIPTION
**Pull request by allnes**
Work with [https://github.com/opencv/opencv_extra/pull/527](https://github.com/opencv/opencv_extra/pull/527)

```
force_builds=Linux OpenCL,Linux AVX2
build_shared:Linux OpenCL=OFF
build_world:Linux AVX2=ON
docker_image:Docs=docs-js
docker_image:Custom=javascript
```

### CODE REVIEW

File: modules/objdetect/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,8 @@
 set(the_description "Object Detection")
-ocv_define_module(objdetect opencv_core opencv_imgproc WRAP java python js)
+ocv_define_module(objdetect opencv_core opencv_imgproc opencv_calib3d WRAP java python js)
+
+if(WITH_QUIRC)
```
Comment (alalek) [inline modules/objdetect/CMakeLists.txt]: HAVE_QUIRC

- `WITH_*` - auto detect build feature, user-defined option
- `HAVE_*` - build feature is available

File: modules/objdetect/src/qrcode.cpp [hunk 2]
```diff
@@ -792,4 +799,268 @@ CV_EXPORTS bool detectQRCode(InputArray in, std::vector<Point> &points, double e
...
+    }
+    return true;
+}
+
+bool QRDecode::release()
+{
+    if (!updatePerspective())  { return false; }
+    if (!versionDefinition())  { return false; }
+    if (!samplingForVersion()) { return false; }
+    if (!decodingProcess())    { return false; }
```
Comment (alalek) [inline modules/objdetect/src/qrcode.cpp]: `release()` methods usually release/free used memory.

File: 3rdparty/quirc/LICENSE [hunk 3]
```diff
@@ -0,0 +1,16 @@
+quirc -- QR-code recognition library
+Copyright (C) 2010-2012 Daniel Beer <dlbeer@gmail.com>
```
Comment (alalek) [inline 3rdparty/quirc/LICENSE]: add `ocv_install_3rdparty_licenses(...)` into CMakeLists.txt

File: modules/objdetect/CMakeLists.txt (hunk 1 above)
Comment (allnes) [reply modules/objdetect/CMakeLists.txt]: @alalek done.

File: modules/objdetect/src/qrcode.cpp (hunk 2 above)
Comment (allnes) [reply modules/objdetect/src/qrcode.cpp]: @alalek done.

File: 3rdparty/quirc/LICENSE (hunk 3 above)
Comment (allnes) [reply 3rdparty/quirc/LICENSE]: @alalek done.

**Comment by allnes** [comment]
@alalek @vpisarev please review me.

**Comment by alalek** [comment]
Don remove assigned builders. If I added JavaScript builder then it should not become broken by this patch.

**Comment by alalek** [comment]
:+1:

**Pull request merged**